### PR TITLE
chore(spring-boot-starter): Set Maven Compiler Level 11

### PR DESCRIPTION
**Pull request by psavidis**
Related-to: https://github.com/camunda/camunda-bpm-platform/issues/3328

### CODE REVIEW

File: spring-boot-starter/example-invoice/pom.xml [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <spring.boot.version>3.1.1</spring.boot.version>
     <graaljs.version>21.1.0</graaljs.version>
     <maven.compiler.plugin.version>3.8.1</maven.compiler.plugin.version>
+    <maven.compiler.release>11</maven.compiler.release>
     <java.version>17</java.version>
```
Comment (danielkelemen) [inline spring-boot-starter/example-invoice/pom.xml:15]: ❓This looks weird. Which one is correct? I think latest Spring version needs 17.

**Review by danielkelemen — COMMENTED** [review]
❓Do we need to update these at all? I think @yanavasileva just recently updated them.

**Comment by yanavasileva** [comment]
Please note, spring boot runs only with JDK 17 (as specified in the Readme). Only improvement that I see here is bump the `maven-compiler-plugin` to `3.11.0` and use `maven.compiler.release` property.

**Comment by yanavasileva** [comment]
Please update the `camunda-version` from `7.20.0-SNAPSHOT` to `7.20.0` as the GHA didn't work after my changes.

File: spring-boot-starter/example-invoice/pom.xml (hunk 1 above)
Comment (psavidis) [reply spring-boot-starter/example-invoice/pom.xml]: The correct value is 17 as you correctly questioned. I've updated all spring boot examples to populate property `maven.compiler.release` with 17 version.

**Review by danielkelemen — APPROVED** [review]
👍 Looks pretty good, left one question about the maven property.

**Review by danielkelemen — APPROVED** [review]
👍

**Comment by psavidis** [comment]
For the sake of completeness, this PR was passing using maven `3.9` but **not** `3.8`. After using maven `3.8`, the `maven-compiler-plugin` declaration is necessary for the build to pass and tested using the latest changes.

**Pull request merged**